op-e2e/setup.go

```diff
@@ -178,135 +280,78 @@ func (cfg SystemConfig) start() (*System, error) {
 		}
 	}()
 
-	// Wallet
-	wallet, err := hdwallet.NewFromMnemonic(cfg.Mnemonic)
-	if err != nil {
-		return nil, fmt.Errorf("Failed to create wallet: %w", err)
-	}
-	sys.wallet = wallet
-
-	// Create the BSS and set it's config here because it needs to be derived from the accounts
-	bssPrivKey, err := wallet.PrivateKey(accounts.Account{
-		URL: accounts.URL{
-			Path: cfg.BatchSubmitterHDPath,
-		},
-	})
+	l1Genesis, err := genesis.BuildL1DeveloperGenesis(cfg.DeployConfig)
 	if err != nil {
 		return nil, err
 	}
-	batchSubmitterAddr := crypto.PubkeyToAddress(bssPrivKey.PublicKey)
 
-	p2pSignerPrivKey, err := wallet.PrivateKey(accounts.Account{
-		URL: accounts.URL{
-			Path: cfg.P2PSignerHDPath,
-		},
-	})
-	if err != nil {
-		return nil, err
+	for addr, amount := range cfg.Premine {
+		existing, ok := l1Genesis.Alloc[addr]
+		if ok {
+			l1Genesis.Alloc[addr] = core.GenesisAccount{
+				Code:    existing.Code,
+				Storage: existing.Storage,
+				Balance: amount,
+				Nonce:   existing.Nonce,
+			}
+		} else {
+			l1Genesis.Alloc[addr] = core.GenesisAccount{
+				Balance: amount,
+				Nonce:   0,
+			}
+		}
 	}
-	p2pSignerAddr := crypto.PubkeyToAddress(p2pSignerPrivKey.PublicKey)
 
-	// Create the L2 Outputsubmitter Address and set it here because it needs to be derived from the accounts
-	l2OOSubmitter, err := wallet.PrivateKey(accounts.Account{
-		URL: accounts.URL{
-			Path: cfg.L2OutputHDPath,
-		},
-	})
+	l1Block := l1Genesis.ToBlock()
+	l2Addrs := &genesis.L2Addresses{
+		ProxyAdmin:                  predeploys.DevProxyAdminAddr,
+		L1StandardBridgeProxy:       predeploys.DevL1StandardBridgeAddr,
+		L1CrossDomainMessengerProxy: predeploys.DevL1CrossDomainMessengerAddr,
+	}
+	l2Genesis, err := genesis.BuildL2DeveloperGenesis(cfg.DeployConfig, l1Block, l2Addrs)
```

Comment:
Curious what you think here, once https://github.com/ethereum-optimism/optimism/pull/3644 is merged then you can pass `nil` for `l2Addrs` and the dev values will be used. I think this is useful because this struct might need to be updated and it'll let us delete the same boilerplate in many places